staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -187,11 +194,23 @@ type WatchServer struct {
 // ServeHTTP serves a series of encoded events via HTTP with Transfer-Encoding: chunked
 // or over a websocket connection.
 func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	returnEarly := true
+	defer func() {
+		if returnEarly {
```

Comment:
For now [given we don't offload to a separate goroutine]:
- let's not move watch ownership to here [as asked in the comment above]
- then for MemoryAllocator you can simply fo:
```
defer func() {
  if s.MemoryAllocator != nil {
    runtime.AllocatorPool.Put(s.MemoryAllocator)
  }
}()
```

Reply:
Done